### ISSUE TITLE
Fix broken GPG magic

### ISSUE THREAD

**Issue by bpsm** [bpsm/edn-java#64]
Whatever magic set of configurations I made years ago to allow releases to Maven Central via Sonatype has ceased functioning for reasons I have not yet been able to diagnose. As a result 0.7.0 has been tagged, but not published on Maven Central.

```
[INFO] --- maven-gpg-plugin:1.6:sign-and-deploy-file (default-cli) @ edn-java ---
gpg: using "........" as default secret key for signing
gpg: signing failed: No pinentry
gpg: signing failed: No pinentry
```

I've googled around for what to do about "No pinetry" but not yet found anything that leads me to solution.

**Comment by aadrian** [comment]
@bpsm https://kemitix.github.io/2018/02/24/gpg-signing-failed.html does not fix the issue?

**Comment by bpsm** [comment]
Thanks.

I'd found something similar yesterday,but it was without `--batch`. I tried adding `--batch` myself after grovelling through the gpg man page, but it didn't seem to help. I'll give it another shot with a fresh release. (probably 0.7.1)

**Comment by bpsm** [comment]
I was able to get deploy to oss.sonatype.org to work as follows:

1. I added the configuration flags as [described above](https://kemitix.github.io/2018/02/24/gpg-signing-failed.html) to pom.xml
2. `mvn release:prepare -Pgpg` activates the gpg profile from my settings.xml which sets gpg.passphrase. Previously I'd not been able to get release:perform to work reliably and had given up on it and was doing releases by hand, followed by explicit calls to gpg:sign-and-deploy-file.
3. `mvn release:perform -Pgpg` works, taking the credentials necessary to upload to oss.sonatype.org form my settings.xml
